eva_sub_cli/validators/validator.py

```diff
@@ -161,63 +164,96 @@ def check_if_file_missing(self):
+                                                           all('evidence_type' in v and v['evidence_type'] is not None
+                                                               for k, v in
+                                                               self.results.get('evidence_type_check', {}).items()
+                                                               if isinstance(v, dict)))
+        elif VCF_CHECK not in self.previous_validation_results:
+            self.results.update({'vcf_check': {'pass': PROCESS_NOT_RUN_YET}})
+            self.results.update({'evidence_type_check': {'pass': PROCESS_NOT_RUN_YET}})
+
+        if ASSEMBLY_CHECK in self.tasks:
+            # assembly_check result
+            asm_nb_mismatch_result = all((asm_check.get('nb_mismatch', 1) == 0
+                                          for vcf_name, asm_check in self.results.get('assembly_check', {}).items()))
+            asm_nb_error_result = all((asm_check.get('nb_error', 1) == 0
+                                       for vcf_name, asm_check in self.results.get('assembly_check', {}).items()))
+            self.results['assembly_check']['pass'] = asm_nb_mismatch_result and asm_nb_error_result
+
+            # fasta_check result
+            fasta_check_result = all((fa_file_check.get('all_insdc', False) is True
+                                      for fa_file, fa_file_check in self.results.get('fasta_check', {}).items()))
+            self.results['fasta_check']['pass'] = fasta_check_result
+        elif ASSEMBLY_CHECK not in self.previous_validation_results:
+            self.results.update({'assembly_check': {'pass': PROCESS_NOT_RUN_YET}})
+            self.results.update({'fasta_check': {'pass': PROCESS_NOT_RUN_YET}})
+
+        if SAMPLE_CHECK in self.tasks:
+            # sample check result
+            self.results['sample_check']['pass'] = self.results.get('sample_check', {}).get('overall_differences',
+                                                                                            True) is False
+        elif SAMPLE_CHECK not in self.previous_validation_results:
+            self.results.update({'sample_check': {'pass': PROCESS_NOT_RUN_YET}})
+
+        if METADATA_CHECK in self.tasks:
+            # metadata check result
+            metadata_xlsx_result = len(self.results.get('metadata_check', {}).get('spreadsheet_errors', []) or []) == 0
+            metadata_json_result = len(self.results.get('metadata_check', {}).get('json_errors', []) or []) == 0
+            self.results['metadata_check']['pass'] = metadata_xlsx_result and metadata_json_result
+        elif METADATA_CHECK not in self.previous_validation_results:
+            self.results.update({'metadata_check': {'pass': PROCESS_NOT_RUN_YET}})
+
+        # take previous result and overwrite with new result
+        self.results = {**self.previous_validation_results, **self.results}
```

Comment:
You can just load previous results into `self.results` directly and modify it directly with this method - no need for the separate `previous_validation_results` variable.

Reply:
updated